Public/Src/FrontEnd/MsBuild/PipConstructor.cs

```diff
@@ -590,11 +595,25 @@ private void SetProcessEnvironmentVariables(IReadOnlyDictionary<string, string>
             //
             // All child processes: Don't wait to kill the processes.
             // CODESYNC: CloudBuild repo TrackerExecutor.cs "info.NestedProcessTerminationTimeout = TimeSpan.Zero"
-            processBuilder.AllowedSurvivingChildProcessNames = ReadOnlyArray<PathAtom>.FromWithoutCopy(
+            var allowedSurvivingChildProcessNames = new List<PathAtom>() { 
                 PathAtom.Create(m_context.StringTable, "mspdbsrv.exe"),
                 PathAtom.Create(m_context.StringTable, "vctip.exe"),
-                PathAtom.Create(m_context.StringTable, "conhost.exe"),
-                PathAtom.Create(m_context.StringTable, "VBCSCompiler.exe"));
+                PathAtom.Create(m_context.StringTable, "conhost.exe")};
+
+            // If the sandbox supports process breakaway and shared compilation is configured to run, we configure VBCSCompiler as such
+            // Otherwise, we add it as a process that is safe to kill when it survives
+            var vbcsCompiler = PathAtom.Create(m_context.StringTable, "VBCSCompiler.exe");
+            if (UseSharedCompilation)

```

Comment:
Wonder if there's a nenvironment or msbuild property to turn shared compilation off as well. If it's turned off, then bxl will duplicate the file accesses right?

Reply:
You mean a mechanism that overrides passing /p:UseSharedCompilation=false to MSBuild?
I was assuming passing that global property will trump any other configuration at this respect. I'm not 100% sure though.
However, even in that case duplicating the accesses is not really a big deal. The cost of actually reporting those accesses should be not significant. And having duplicates is not a problem.

---
In reply to: [346069926](https://github.com/microsoft/BuildXL/pull/1207#discussion_r346069926) [](ancestors = 346069926)

Reply:
The msbuild build could decide to turn off shared compilation (e.g. turning it off in a `directory.build.props`), in which case bxl can skip shared compilation as well. Or maybe that already happens and I missed it.

Reply:
But if there's no real consequence on logging the IO regardless of what msbuild chooses to do then it doesn't matter.
Also, there could be build time logic to change shared compilation, so I guess it's not really knowable what the build decides to do